### ISSUE TITLE
Install TestRemovePath test is flaky on Windows

### ISSUE THREAD

**Issue by cmacknz** [elastic/elastic-agent#3055]
## Flaky Test

* **Test Name:** Test / Matrix - PLATFORM = 'windows-2016 && windows-immutable' / Test / github.com/elastic/elastic-agent/internal/pkg/agent/install.TestRemovePath (from github.com_elastic_elastic-agent_internal_pkg_agent_install
* **Link:** https://github.com/elastic/elastic-agent/blob/231c53612bb35ce2c7feb1c73e7b9351405d3fa3/internal/pkg/agent/install/uninstall_windows_test.go#L53-L54
* **Branch:** 8.9 originally in https://github.com/elastic/elastic-agent/pull/3054
* **Artifact Link:** https://fleet-ci.elastic.co/job/elastic-agent/job/elastic-agent-mbp/job/PR-3054/1/testReport/junit/github/com_elastic_elastic-agent_internal_pkg_agent_install/Test___Matrix___PLATFORM____windows_2016____windows_immutable____Test___TestRemovePath/
* **Notes:** I suspect this is a new failure as a consequence of https://github.com/elastic/elastic-agent/pull/3014

### Stack Trace

```
=== RUN   TestRemovePath
    uninstall_windows_test.go:54: 
        	Error Trace:	C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-3054/src/github.com/elastic/elastic-agent/internal/pkg/agent/install/uninstall_windows_test.go:54
        	Error:      	Received unexpected error:
        	            	failed to open handle for "C:\\Users\\jenkins\\AppData\\Local\\Temp\\TestRemovePath396485393\\001\\subdir\\main.exe": The process cannot access the file because it is being used by another process.
        	Test:       	TestRemovePath
--- FAIL: TestRemovePath (3.78s)
```

**Comment by cmacknz** [comment]
I reverted the change to the Windows uninstall command in 8.9 in https://github.com/elastic/elastic-agent/pull/3056 until we can root cause why the test failed. We can store this in 8.9 once we have a fix.

**Comment by cmacknz** [comment]
From https://github.com/elastic/elastic-agent/issues/3069 this test can also fail because the test program is identified as malicious 🤦 

```
[2023-07-12T13:23:48.438Z] === FAIL: internal/pkg/agent/install TestRemovePath (2.58s)
[2023-07-12T13:23:48.438Z]     uninstall_windows_test.go:47: 
[2023-07-12T13:23:48.438Z]         	Error Trace:	C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-2781/src/github.com/elastic/elastic-agent/internal/pkg/agent/install/uninstall_windows_test.go:47
[2023-07-12T13:23:48.438Z]         	Error:      	Received unexpected error:
[2023-07-12T13:23:48.438Z]         	            	fork/exec C:\Users\jenkins\AppData\Local\Temp\TestRemovePath2314055137\001\subdir\main.exe: Operation did not complete successfully because the file contains a virus or potentially unwanted software.
[2023-07-12T13:23:48.438Z]         	Test:       	TestRemovePath
```

**Comment by cmacknz** [comment]
Reverting this on main as well as it is causing random test failures in CI.

**Comment by blakerouse** [comment]
Fixed in #3120
